fractal_server/app/runner/v2/__init__.py

```diff
@@ -137,8 +138,8 @@ async def submit_workflow(
             )
             return
 
-        dataset: DatasetV2 = db_sync.get(DatasetV2, dataset_id)
-        workflow: WorkflowV2 = db_sync.get(WorkflowV2, workflow_id)
+        dataset: Optional[DatasetV2] = db_sync.get(DatasetV2, dataset_id)
```

Comment:
Move up and ref https://github.com/fractal-analytics-platform/fractal-server/issues/1709